core/node/events/miniblock_info.go

```diff
@@ -497,3 +497,21 @@ func (p *ParsedMiniblockInfoOpts) WithApplyOnlyMatchingSnapshot() *ParsedMiniblo
 func (p *ParsedMiniblockInfoOpts) ApplyOnlyMatchingSnapshot() bool {
 	return p.applyOnlyMatchingSnapshot
 }
+
+func MiniblockInfosToStorageMbs(mbs []*MiniblockInfo) ([]*storage.MiniblockDescriptor, error) {
+	storageMbs := make([]*storage.MiniblockDescriptor, len(mbs))
+	var err error
+	for i, mb := range mbs {
+		if i > 0 && mb.Ref.Num != mbs[0].Ref.Num+int64(i) {
```

Comment:
Can we also check ensure that the header prev mb hash points to the previous block to ensure that blocks are not only sequel but also form a canonical mb chain?